### ISSUE TITLE
[Python] Support for DSPy

### ISSUE THREAD

**Issue by mikeldking** [Arize-ai/openinference#109]
Add DSPy tracing for observability

**Comment by thomasahle** [comment]
I'm afraid we made a few changes to DSpy that breaks `instrumentation/dspy
/__init__.py`.

For example, the loop
```python
for field in signature.fields:
    if field.output_variable and field.output_variable in prediction:
        output[field.output_variable] = prediction.get(field.output_variable)
```
Should now be
```python
for key, field in signature.fields.items():
    if key in prediction:
        output[key] = prediction.get(key)
```

If anything else is broken, please let me know, and I will happily help fix it.

**Comment by axiomofjoy** [comment]
Thanks @thomasahle!

**Comment by sandangel** [comment]
<img width="1587" alt="image" src="https://github.com/Arize-ai/openinference/assets/22189661/a243404d-ecd8-47f1-a35a-8022d205174f">

I got an error with dspy.Google. I can see it's success if using dspy.OllamaLocal. Is there a way to test this working for all dspy providers?

**Comment by mikeldking** [comment]
Good catch @sandangel - we'll get our coverage expanded on DSPy LMs - you can track the issue here: https://github.com/Arize-ai/openinference/issues/254

**Comment by axiomofjoy** [comment]
> <img alt="image" width="1587" src="https://private-user-images.githubusercontent.com/22189661/309967890-a243404d-ecd8-47f1-a35a-8022d205174f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk3NTk5MDUsIm5iZiI6MTcwOTc1OTYwNSwicGF0aCI6Ii8yMjE4OTY2MS8zMDk5Njc4OTAtYTI0MzQwNGQtZWNkOC00N2YxLWEzNWEtODAyMmQyMDUxNzRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA2VDIxMTMyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEwN2RjOWE4YWJmY2I5NGFiNTFmYzFhMWI2NjlmN2UzMDU1YWE3MWJjYWY2ZGRiYTNhM2NjYWM5OGY1MTAwMDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.1aOzgOAbCaac9mB5yXq21uRHWXAgfabiA-_KxRPVlMc">
> After updating to 0.1.2, I got an error with dspy.Google. I can see it's success if using dspy.OllamaLocal. Is there a way to test this working for all dspy providers?

@sandangel This issue is resolved in `openinference-instrumentation-dspy==0.1.3`. Thanks for the bug report!
